scheduler/mesos/mesos.go

```diff
@@ -222,56 +235,71 @@ func (m *mesosScheduler) getMesosMaster() ([]string, string, error) {
 	return masterHosts, protocol, nil
 }
 
-func (m *mesosScheduler) getMesosTask(taskId string) (mesosTask, string, string, error) {
-	var state mesosState
-	var masterErr error
-	if masterHosts, protocol, err := m.getMesosMaster(); err == nil {
-		for _, host := range masterHosts {
-			if resp, err := http.Get(protocol + "://" + host + "/state.json"); err == nil {
-				defer resp.Body.Close()
-				if err := json.NewDecoder(resp.Body).Decode(&state); err == nil {
-					if state.Pid == state.Leader {
-						masterErr = nil
-						break
-					}
-				} else {
-					masterErr = err
+func (m *mesosScheduler) getMesosTask(taskID string) (mesosTask, string, string, error) {
+	var tasks mesosTasks
+	var err error
+	if err := m.sendMesosRequest("/tasks?task_id="+taskID, "", "GET", &tasks); err == nil {
+		if len(tasks.Tasks) != 1 {
+			return mesosTask{}, "", "", scheduler.ErrTaskNotFound
+		}
+		task := tasks.Tasks[0]
+		if slaveHost, err := m.getSlaveHost(task.SlaveId); err == nil {
+			if frameworkName, ok := m.frameworks[task.FrameworkID]; ok {
+				return task, frameworkName, slaveHost, nil
+			} else if err := m.updateFrameworks(); err == nil {
```

Comment:
I'm thinking we should probably log here indicating that there was a cache miss and on what FrameworkID.